pkg/pgmodel/upgrade_tests/upgrade_test.go

```diff
@@ -159,89 +112,48 @@ func TestUpgradeFromPrev(t *testing.T) {
 	//      gathered the db info in parallel. Unfortunately our db runner doesn't
 	//      support this yet
 	var upgradedDbInfo dbInfo
-	withNewDB(t, cleanImage, prevVersion.String(), *testDatabase,
-		func(container testcontainers.Container, connectURL string, db *pgxpool.Pool, tmpDir string) {
-			ingestor, err := pgmodel.NewPgxIngestor(db)
-			if err != nil {
-				t.Fatalf("error connecting to DB: %v", err)
-			}
-			_, err = ingestor.Ingest(copyMetrics(preUpgradeData1), &prompb.WriteRequest{})
-			if err != nil {
-				t.Fatalf("ingest error: %v", err)
-			}
-			_ = ingestor.CompleteMetricCreation()
-			_, err = ingestor.Ingest(copyMetrics(preUpgradeData2), &prompb.WriteRequest{})
-			if err != nil {
-				t.Fatalf("ingest error: %v", err)
-			}
-			_ = ingestor.CompleteMetricCreation()
-			ingestor.Close()
-			defer db.Close()
+	withDBAndConnector(t, *testDatabase, "timescale/timescale-prometheus:"+prevVersion.String(),
```

Comment:
can we break this out into getUpgradedDbInfo() DbInfo

Reply:
We should also document what this function is doing

Reply:
what's "this" here?